### PR TITLE
Converting ConversationItem to MVVM

### PR DESCRIPTION
**Pull request by thecombatwombat**
This PR should be rebased and merged after #218 is merged, since changes there will have to be reflected here.

### CODE REVIEW

File: layer-atlas/src/main/java/com/layer/ui/MessageComposer.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ public MessageComposer setTextSender(TextSender textSender) {
     public MessageComposer addAttachmentSenders(AttachmentSender... senders) {
         for (AttachmentSender sender : senders) {
             if (sender.getTitle() == null && sender.getIcon() == null) {
-                throw new NullPointerException("Attachment handlers must have at least a title or icon specified.");
+                throw new NullPointerException("Attachment handlers must have at least a mTitle or icon specified.");
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/MessageComposer.java]: The `NullPointerException` will be logged anyway...

File: layer-atlas/src/main/java/com/layer/ui/ConversationsRecyclerView.java [hunk 2]
```diff
@@ -25,22 +25,33 @@
...
+import java.util.Set;
+
 public class ConversationsRecyclerView extends RecyclerView {
     ConversationsAdapter mAdapter;
     private ItemTouchHelper mSwipeItemTouchHelper;
-
     private ConversationStyle conversationStyle;
+    private Set<CellFactory> mDefaultCellFactories;
+    private LayerClient mLayerClient;
+    private Picasso mPicasso;
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/ConversationsRecyclerView.java]: I strongly prefer to merge this after the Avatar stuff is merged in to `layer_ui` and I can rebase off that.

File: layer-atlas/src/main/res/layout/new_ui_conversation_item.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+    <LinearLayout
```
Comment (thecombatwombat) [inline layer-atlas/src/main/res/layout/new_ui_conversation_item.xml]: This class is to be removed

File: layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import com.layer.sdk.messaging.Identity;
+import com.layer.ui.util.ConversationFormatter;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ConversationItemViewModel extends BaseObservable {
+    //View Logic
+    protected final ConversationFormatter mConversationFormatter;
+    protected LayerClient mLayerClient;
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java]: We use this only for authenticated user. Let's pass that in. Rename and add a param to `setConversation`?

File: layer-atlas/src/main/java/com/layer/ui/conversation/OnConversationItemClickListener.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.layer.ui.conversation;
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/conversation/OnConversationItemClickListener.java]: Should the package be called `conversationItem`?

File: layer-atlas/src/main/java/com/layer/ui/binding/Bindings.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.layer.ui.binding;
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/binding/Bindings.java]: Remove this if not used. We can add this later if needed.

File: layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+    // View Data
+    public Conversation mConversation;
+    protected Set<Identity> mParticipants;
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java]: Rename `mParticipants` to `mParticipantsMinusCurrentUser`

File: layer-atlas/src/main/java/com/layer/ui/util/ConversationFormatter.java [hunk 8]
```diff
@@ -1,24 +1,54 @@
...
+import java.util.Calendar;
+import java.util.Date;
 import java.util.Set;
 
 /**
- * A formatter that enables us to modify  conversation recyclerView items in the UI
+ * A formatter that enables us to modify conversation recyclerView items in the UI
  */
 
 public class ConversationFormatter {
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/util/ConversationFormatter.java]: Rename to `ConversationItemFormatter` ?

File: layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java (hunk 4 above)
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java]: Done

File: layer-atlas/src/main/java/com/layer/ui/binding/Bindings.java (hunk 6 above)
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/binding/Bindings.java]: I stand corrected. This is used in `ui_conversation_item.xml` bindings for setting the typeface on a `TextView`

File: layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java (hunk 7 above)
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/conversation/ConversationItemViewModel.java]: Done

File: layer-atlas/src/main/java/com/layer/ui/util/ConversationFormatter.java (hunk 8 above)
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/util/ConversationFormatter.java]: Done

File: layer-atlas/src/main/res/layout/new_ui_conversation_item.xml (hunk 3 above)
Comment (thecombatwombat) [inline layer-atlas/src/main/res/layout/new_ui_conversation_item.xml]: Done

**Pull request merged**